### ISSUE TITLE
"Bogus virtual array access" on save after from_dct with some picture dimensions and 4:2:0 chroma subsampling

### ISSUE THREAD

**Issue by Masstock** [martinbenes1996/jpeglib#9]
Hello !
First of all, thanks a lot for this usefull library that I use for my experiments.

As said in the title, I have issues with some picture dimensions and specific chroma subsampling formats. For exemple, a 800x600 picture with 4:2:0 subsampling produce a YCbCr DCT blocks array with shape `((75, 100), (38, 50), (38, 50))`. But at saving, the `samp_factor` function in `_infere.py` produce following factor array:

```
array([[1.97368421, 2.        ],
       [1.        , 1.        ],
       [1.        , 1.        ]])
```

Which is truncated to `int16` (causing the first value to be rounded down to 1 rather than 2). This cause an error `"Bogus virtual array access"` at generated `DCTJPEG` object saving. I made an attempt to fix this issue by adding a `np.ceil` function call in `samp_factor()`.

```py
factor = np.ceil(np.array([
    max_subs,
    *(max_subs / dims)
])).astype('int16')
```

See my fork where I commit the proposed patch [https://github.com/Masstock/jpeglib](url).
I hope I did well, thanks !

**Comment by martinbenes1996** [comment]
A good spot. Will take a look into that. Can you create pull request?

**Comment by Masstock** [comment]
Sure, I try to do so.

**Comment by martinbenes1996** [comment]
Thank you, merged to dev now.

I added a new unit test for this bug based on your report.
https://github.com/martinbenes1996/jpeglib/commit/ed34f228e5ab0f33e49c742de306f9d6623f7ece

The fix will go to the package with the new release.

**Comment by martinbenes1996** [comment]
The fix is part of release 0.12.7.
https://github.com/martinbenes1996/jpeglib/releases/tag/0.12.7
